Fix unhandled error on client from SettingOption.ts

richardFocus opened the pull request:
## Relevant components:
- [ ] Signalling server
- [ ] Common library
- [X] Frontend library
- [ ] Frontend UI library
- [ ] Matchmaker
- [ ] Platform scripts
- [ ] SFU

## Problem statement:
I have been having unhandled errors in production and found this update solves the issue we were having originating from SettingOptions.ts:
![settingOptionsError](https://github.com/user-attachments/assets/ab47ca26-3478-4f7b-98d2-6a0ad4cf98f0)


## Solution
The was an issue with the options not being ready in time, I am not sure if this is the best way to solve the bug as it may be better to solve elsewhere in the codebase but this has fixed the issue for us and now I don't get anymore production errors, if someone else knows a better fix then go for it.

## Testing
I have been running this code in productions for weeks now and no more issues that were occurring previously at least every few days.

richardFocus committed:
Update SettingOption.ts

mcottontensor committed:
Fixed linting

lukehb reviewed (changes requested):
The change to add some error handling seem okay.

@mcottontensor I think the logging to console should use our logging class instead, right?

lukehb commented on Frontend/library/src/Config/SettingOption.ts:
```diff
@@ -74,18 +74,32 @@ export class SettingOption<CustomIds extends string = OptionParametersIds> exten
-        filteredList = this.options.filter((option: string) => option.indexOf(value.split(' ')[0]) !== -1);
-        if (filteredList.length) {
+        // If no match is found, and value contains parameters, try matching only the codec
+        const valueCodec = value.split(' ')[0];
```
`value.split(' ')[0]` could itself fail (same with above on line 83) - I would request some defensive programming be done around that too, e.g. check if split results in a non empty array, then that value [0] is non-empty.

richardFocus replied:
Yes you are correct.

lukehb reviewed (changes requested):
Some further changes come to mind making it even safer.

richardFocus committed:
Update SettingOption.ts

Updated console warning to use local Logger function and added conditional check filter function as requested.

richardFocus reviewed (commented):
Updated as requested.

richardFocus committed:
Update SettingOption.ts

Updated import path of Logger function

richardFocus reviewed (commented):
Updated import path for Logger function

richardFocus committed:
Update SettingOption.ts

Removed getStrackTrace()

richardFocus reviewed (commented):
Removed getStrackTrace()

richardFocus committed:
Update SettingOption.ts

linting fix

richardFocus reviewed (commented):
linting fix

richardFocus committed:
Update SettingOption.ts

fixed linting

richardFocus reviewed (commented):
fixed linting
